### ISSUE TITLE
Incorrect behavior of SimplifyDeMorganBinaryRector

### ISSUE THREAD

**Issue by zingimmick** [rectorphp/rector#5153]
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

Negated conditions was changed from `!($c >= $a + $b)` to `!$c < !($a + $b)`.

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/68a3508e-c718-4f99-86ed-884d617e7e69

```php
<?php

function demo(int $a, int $b, int $c): bool
{
    if (! ($c >= $a + $b)) {
        return true;
    }
    return false;
}
```

### Responsible rules

* `SimplifyDeMorganBinaryRector`

## Expected Behavior

Nothing needs to be changed

**Comment by zingimmick** [comment]
PR #5164 only fix on negated greater or equal, but the behavior of other binary operators still incorrect.

See https://getrector.org/demo/0bc7409d-bf13-4f32-a5c3-ca0038fbc0b6

```php
<?php

function demo(): void
{
    $a = 1;
    $b = 10;
    $c = 1000;
    $d = 100;

    // keep it
    $e = ! ($a >= $b);
    $f = ! ($a <= $b);
    $g = ! ($a > $b);
    $h = ! ($a < $b);

    // should not skip
    $i = ! ($a > $b or $c <= $d);
}
```

May be only `BooleanOr` and `LogicalOr` needs to be changed?

**Comment by samsonasik** [comment]
Fixed at #5177
